src/backend/marsha/core/serializers/video.py

```diff
@@ -396,40 +406,56 @@ def validate(self, attrs):
                                 "authentication."
                             }
                         )
+                # We can identify the user for this context_id and consumer_site, we make sure
+                # this user hasn't already registered for this video.
+                if LiveRegistration.objects.filter(
+                    consumer_site=attrs["consumer_site"],
+                    deleted=None,
+                    lti_id=attrs["lti_id"],
+                    lti_user_id=attrs["lti_user_id"],
+                    video=video,
+                ).exists():
+                    raise serializers.ValidationError(
+                        {
+                            "lti_user_id": "This identified user is already "
+                            "registered for this video and consumer site and "
+                            "course."
+                        }
+                    )
 
-                # We can identify the user for this context_id, we make sure this user hasn't
-                # already registered for this video. It's only relevant if context_id is defined.
+            else:  # public token should have no LTI info
                 if (
                     user.token.payload.get("context_id")
-                    and LiveRegistration.objects.filter(
-                        consumer_site=attrs["consumer_site"],
-                        deleted=None,
-                        lti_user_id=attrs["lti_user_id"],
-                        video=video,
-                    ).exists()
+                    or user.token.payload.get("consumer_site")
+                    or (
+                        user.token.payload.get("user")
+                        and user.token.payload["user"].get("id")
+                    )
                 ):
+                    # we prevent any side effects if token's creation changes.
                     raise serializers.ValidationError(
                         {
-                            "lti_user_id": "This identified user is already "
-                            "registered for this video and consumer site."
+                            "token": "Public token shouldn't have any LTI information, "
+                            "cases are not expected and coded."
```

Comment:
```suggestion
                            "cases are not expected."
```

`coded` should be replaced by `implemented` but this info should not be revealed IMO